apps/sms/views/conversation/js/conversation.js

```diff
@@ -267,48 +267,44 @@ var ConversationView = {
     this.onMessageTypeChange = this.onMessageTypeChange.bind(this);
   },
 
-  onVisibilityChange: function conv_onVisibilityChange(e) {
-    // If we leave the app and are in a thread or compose window
-    // save a message draft if necessary
-    if (document.hidden) {
-      // Auto-save draft if the user has entered anything
-      // in the composer or into To field (for composer panel only).
-      var isAutoSaveRequired = false;
+  onVisibilityChange: function conv_onVisibilityChange() {
+    // If we leave the app and are in a thread or compose window update a
+    // message draft if necessary.
+    var isDraftUpdateNeeded = document.hidden &&
+      (Navigation.isCurrentPanel('composer') ||
+      Navigation.isCurrentPanel('thread'));
 
-      if (Navigation.isCurrentPanel('composer')) {
-        isAutoSaveRequired = !Compose.isEmpty() ||
-          !!ConversationView.recipients.length;
-      } else if (Navigation.isCurrentPanel('thread')) {
-        isAutoSaveRequired = !Compose.isEmpty();
-      }
-
-      if (isAutoSaveRequired) {
-        ConversationView.saveDraft(true /* preserveDraft */);
-      }
+    if (isDraftUpdateNeeded) {
+      this.updateDraft();
```

Comment:
One thing changed here, previously we never discarded draft on visibility change, we were only saving. With this code, if message content is cleared and then hidden, draft if any will be discarded. At first I had mixed feelings, but decided to do it for the consistency in the end. What do you think?